fix(cohorts): apply HogQL global settings to cohort queries

haacked opened the pull request:
Cohort queries were missing critical settings like transform_null_in=1, causing funnel-to-cohort conversions to return empty results. Now cohort queries use the same settings as regular HogQL queries.

## Problem

Save as Cohort wasn't working for some funnel insights. See https://posthoghelp.zendesk.com/agent/tickets/34771 for an example.

## Changes

Pass `HogQLGlobalSettings()` to `print_cohort_hogql_query`

## How did you test this code?

Ran unit tests.
Manually tested. I wasn't able to confirm fix because I couldn't repro issue locally.

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

haacked committed:
fix: apply HogQL global settings to cohort queries

Cohort queries were missing critical settings like transform_null_in=1,
causing funnel-to-cohort conversions to return empty results. Now
cohort queries use the same settings as regular HogQL queries.

aspicer reviewed (approved):
Looks good, although copilot says the import is missing.

Would probably be good to add a test too to avoid regressions

haacked commented on posthog/models/cohort/util.py:
```diff
@@ -105,7 +105,10 @@ def print_cohort_hogql_query(cohort: Cohort, hogql_context: HogQLContext, *, tea
-    return print_ast(query, context=hogql_context, dialect="clickhouse")
+
+    # Apply HogQL global settings to ensure consistency with regular queries
+    settings = HogQLGlobalSettings()
```
Copilot, you're drunk. It's on line 18:

```python
from posthog.hogql.constants import LimitContext, HogQLGlobalSettings
```

haacked commented:
> Looks good, although copilot says the import is missing.

The import is there on line 18.

> Would probably be good to add a test too to avoid regressions

Good call!

haacked committed:
test: add test for cohort query settings

Adds a test that verifies cohort queries include HogQL global settings.
This test would have failed before the fix and now passes, ensuring
the bug doesn't regress.
